packages/strapi-plugin-graphql/services/GraphQL.js

```diff
@@ -17,6 +17,424 @@ const GraphQLJSON = require('graphql-type-json');
 const policyUtils = require('strapi-utils').policy;
 
 module.exports = {
+  /**
+   * Returns all fields of type primitive
+   * 
+   * @returns {Boolean}
+   */
+  isPrimitiveType: (_type) => {
+    const type = _type.replace('!', '');
+    return (
+      type === 'Int' ||
+      type === 'Float' ||
+      type === 'String' ||
+      type === 'Boolean'
+    );
+  },
+
+  /**
+   * Returns all fields that are not of type array
+   * 
+   * @returns {Boolean}
+   * 
+   * @example
+   * 
+   * isNotOfTypeArray([String])
+   * // => false
+   * isNotOfTypeArray(String!)
+   * // => true
+   */
+  isNotOfTypeArray: (_type) => {
+    const arrayRegexp = /(\[\w+\])/;
+    const type = _type.replace('!', '');
+    return !arrayRegexp.test(type);
+  },
+
+  /**
+   * Convert non-primitive type to string (non-primitive types corresponds to a reference to an other model)
+   * 
+   * @returns {String}
+   */
+  extractType: function (_type) {
+    return this.isPrimitiveType(_type)
+      ? _type.replace('!', '')
+      : 'ID';
```

Comment:
What about Enum type ?
Actually it's considered as an ID (and the same for DateTime scalar)

Reply:
We need to ignore all non-primitive types. Good catch :+1:

Reply:
Now it's working with fields of type enum and date
![image](https://user-images.githubusercontent.com/14129459/41830693-bf9bdcbe-7842-11e8-9ffc-5986ccb41ed4.png)